scrapper.py

```diff
@@ -44,20 +75,106 @@ def get_search_urls(self):
+
+    def parse(self):
+        """
+        Extracts all necessary data from the article web page
+        """
+        response = requests.get(self.article_url, HEADERS)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        """
+        Fills the Article instance with text
+        """
+        content_bs = article_bs.find_all('div', class_="clearfix text-formatted")[1]
+        raw_download_page = content_bs.find_all('p')[1].find('a')['href']
+        download_page = f'https:{raw_download_page}'
+        pdf = PDFRawFile(download_page, self.article_id)
+        pdf.download()
+        self.article.text = pdf.get_text()
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        Fills the Article instance with meta information
+        """
+        journal_title = article_bs.find('span',
+                                        class_="field field--name-title field--type-string field--label-hidden")
+        self.article.title = journal_title.text
+        article_title = article_bs.find_all('div', class_="clearfix text-formatted")[1]
+        links_bs = article_title.find_all('li')
+        for link in links_bs:
+            author_bald = link.find('b')
+            author_strong = link.find('strong')
+
+            if author_bald:
+                self.article.author = author_bald.text
+            elif author_strong:
+                self.article.author = author_strong.text
+            else:
+                self.article.author = 'NOT FOUND'
+
+            if link.find("a")["href"][-4:] == ".pdf":
+                self.article.url = f'https:{link.find("a")["href"]}'
+        date_raw = article_bs.find('span',
+                                   class_="field field--name-created field--type-created field--label-hidden").text[4:]
+        article_date = datetime.datetime.strptime(date_raw, '%d.%m.%Y - %H:%M')
+        self.article.date = article_date
+
+
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    base_path.mkdir(exist_ok=True, parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as file:
+        config = json.load(file)
+
+    seed_urls = config['seed_urls']
+    max_articles = config['total_articles_to_find_and_parse']
+
+    if not seed_urls:
+        raise IncorrectURLError
+
+    http_pattern = re.compile(r"https://iling-ran.ru/web/ru/publications")
+    for url in seed_urls:
+        if not re.match(http_pattern, url):
+            raise IncorrectURLError
+    if not isinstance(max_articles, int) or max_articles <= 0:
```

Comment:
in case max_articles is an integer, TypeError will be thrown before IncorrectNumberOfArticlesError
separate these conditions

Reply:
done!